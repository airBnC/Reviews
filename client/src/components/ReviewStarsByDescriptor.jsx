import React from 'react';
import PropTypes from 'prop-types';

import StarsByDescriptor from './StarsByDescriptor';

import {
  ReviewStarsDescriptorWrapper,
  ReviewStarsDescriptorColumnWrapper,
  ReviewStarsDescriptorItemWrapper,
  ReviewStarsDescriptorTextWrapper,
} from './styles/ReviewStarsByDescriptor.styles';

const ReviewStarsByDescriptor = props => (
  <ReviewStarsDescriptorWrapper>
    <ReviewStarsDescriptorColumnWrapper>
      <ReviewStarsDescriptorItemWrapper>
        <ReviewStarsDescriptorTextWrapper> Accuracy </ReviewStarsDescriptorTextWrapper>
        <StarsByDescriptor avg={props.accuracyAvg} />
      </ReviewStarsDescriptorItemWrapper>
      <ReviewStarsDescriptorItemWrapper>
        <ReviewStarsDescriptorTextWrapper> Communication </ReviewStarsDescriptorTextWrapper>
        <StarsByDescriptor avg={props.communicationAvg} />
      </ReviewStarsDescriptorItemWrapper>
      <ReviewStarsDescriptorItemWrapper>
        <ReviewStarsDescriptorTextWrapper> Cleanliness </ReviewStarsDescriptorTextWrapper>
        <StarsByDescriptor avg={props.cleanlinessAvg} />
      </ReviewStarsDescriptorItemWrapper>
    </ReviewStarsDescriptorColumnWrapper>
    <ReviewStarsDescriptorColumnWrapper>
      <ReviewStarsDescriptorItemWrapper>
        <ReviewStarsDescriptorTextWrapper> Location </ReviewStarsDescriptorTextWrapper>
        <StarsByDescriptor avg={props.locationAvg} />
      </ReviewStarsDescriptorItemWrapper>
      <ReviewStarsDescriptorItemWrapper>
        <ReviewStarsDescriptorTextWrapper> Check-in </ReviewStarsDescriptorTextWrapper>
        <StarsByDescriptor avg={props.checkinAvg} />
      </ReviewStarsDescriptorItemWrapper>
      <ReviewStarsDescriptorItemWrapper>
        <ReviewStarsDescriptorTextWrapper> Value </ReviewStarsDescriptorTextWrapper>
        <StarsByDescriptor avg={props.valueAvg} />
      </ReviewStarsDescriptorItemWrapper>
    </ReviewStarsDescriptorColumnWrapper>
  </ReviewStarsDescriptorWrapper>
);

ReviewStarsByDescriptor.propTypes = {
  accuracyAvg: PropTypes.number,
  communicationAvg: PropTypes.number,
  cleanlinessAvg: PropTypes.number,
  locationAvg: PropTypes.number,
  checkinAvg: PropTypes.number,
  valueAvg: PropTypes.number,
};

ReviewStarsByDescriptor.defaultProps = {
  accuracyAvg: 1,
  communicationAvg: 1,
  cleanlinessAvg: 1,
  locationAvg: 1,
  checkinAvg: 1,
  valueAvg: 1,
};

export default ReviewStarsByDescriptor;