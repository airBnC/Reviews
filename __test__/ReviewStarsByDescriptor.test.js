import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ReviewStarsByDescriptor from '../client/src/components/ReviewStarsByDescriptor';

configure({ adapter: new Adapter() });

describe('ReviewStarsByDescriptor', () => {
  it('should render correctly', () => {
    const output = shallow(<ReviewStarsByDescriptor />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
