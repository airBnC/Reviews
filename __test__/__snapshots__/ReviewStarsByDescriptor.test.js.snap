// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewStarsByDescriptor should render correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.span>
         Accuracy 
      </styled.span>
      <StarsByDescriptor
        avg={1}
      />
    </styled.div>
    <styled.div>
      <styled.span>
         Communication 
      </styled.span>
      <StarsByDescriptor
        avg={1}
      />
    </styled.div>
    <styled.div>
      <styled.span>
         Cleanliness 
      </styled.span>
      <StarsByDescriptor
        avg={1}
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.span>
         Location 
      </styled.span>
      <StarsByDescriptor
        avg={1}
      />
    </styled.div>
    <styled.div>
      <styled.span>
         Check-in 
      </styled.span>
      <StarsByDescriptor
        avg={1}
      />
    </styled.div>
    <styled.div>
      <styled.span>
         Value 
      </styled.span>
      <StarsByDescriptor
        avg={1}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
